packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -909,16 +914,17 @@ export class AnchorSet implements AnchorLocator {
 			},
 			exitNode(index: number): void {
 				assert(this.parent !== undefined, 0x3ac /* Must have parent node */);
-				this.maybeWithNode((p) => {
-					p.events.emit("subtreeChanged", p);
-					if (this.depthThresholdForSubtreeChanged === this.currentDepth) {
+				if (this.depthThresholdForSubtreeChanged === this.currentDepth) {
+					this.maybeWithNode((p) => {
+						p.events.emit("subtreeChanged", p);
+
 						this.bufferedEvents.push({
 							node: p,
 							event: "subtreeChangedAfterBatch",
 						});
-						this.depthThresholdForSubtreeChanged--;
-					}
-				});
+					});
+					this.depthThresholdForSubtreeChanged--;
```

Comment:
Ah, decrementing this wasn't happening if the node hadn't been accessed and `maybeWithNode()...` did nothing, correct? So the after-batch event and its consequences got messed up.

I'm not sure about the other bit though; `subtreeChanged` used to be emitted regardless of the condition and now it only happens when it's true. Is that correct?

Reply:
> I'm not sure about the other bit though; subtreeChanged used to be emitted regardless of the condition and now it only happens when it's true. Is that correct?

Yes, that is correct. That change isn't particularly important (I don't think we use that event currently) and it was valid to have it fire when it did, I figured since we optimized subtreeChangedAfterBatch, we might as well use that same logic to optimize subtreeChanged so we don't fire it way more than we need to and its simpler to code it this way anyway.